theano/gof/opt.py

```diff
@@ -2053,12 +2014,39 @@ def print_profile(stream, prof, level=0):
         print(blanc, "  loop time", loop_t, file=stream)
         print(blanc, "  callback_time", callback_time, file=stream)
         if isinstance(lopt, LocalOptGroup):
-            lopt.print_profile(stream, (lopt.time_opts,
```

Comment:
I don't think having the print_profile can be the problem. But maybe keeping the results of profiling. Can you modify the method the execute the opt to return None instead of the profiling information? This will remove the information saved. If this is the extra information that raise the memory usage, then we will see the cause.